### PR TITLE
Fix signing

### PR DESCRIPTION
**Pull request by pmengelbert**
- Additionally fixes the incorrect storage container name for the "latest" API update.

### CODE REVIEW

File: templates/sign.steps.windows.yml [hunk 1]
```diff
@@ -11,9 +10,13 @@ steps:
...
     tree ${ARTIFACTS_DIR}
 
-    readarray -d '' artifacts < <(find ${ARTIFACTS_DIR} -type f -regextype posix-egrep -regex '.*\.zip$' -print0)
-    for i in "${artifacts[@]}"; do
-      unzip -d $(dirname "$i") $i
+    readarray -d '' artifacts < <(find ${ARTIFACTS_DIR} -maxdepth 1 -type f -regextype posix-egrep -regex '.*\.zip$' -print0)
+
+    mkdir "${ARTIFACTS_DIR}/extracted"
+    for art in "${artifacts[@]}"; do
+      dir="$ARTIFACTS_DIR/extracted/$(basename "$art")"
```
Comment (cpuguy83) [inline templates/sign.steps.windows.yml:17]:
```suggestion
      base="$(basename "${art}")"
      dir="${ARTIFACTS_DIR}/extracted/$(basename "$art")"
```

File: templates/sign.steps.windows.yml [hunk 2]
```diff
@@ -70,17 +73,21 @@ steps:
...
       fi
 
-      zip -j -r "${zip_file}" "${i}" -x *.zip
-      for j in $(ls ${i}/*.exe); do
-          rm ${j}
-      done
+      zip -j -r "${zip_file}" "${dir}"
     done
+
+    rm -rf "$ARTIFACTS_DIR/extracted"
```
Comment (cpuguy83) [inline templates/sign.steps.windows.yml:90]:
```suggestion
    rm -rf "${ARTIFACTS_DIR}/extracted"
```

File: templates/sign.steps.windows.yml [hunk 3]
```diff
@@ -70,17 +73,21 @@ steps:
       ]
 - script: |
     set -o xtrace
-    for i in ${ARTIFACTS_DIR}/windows/*; do
-      zip_file="$(file ${i}/*.zip | cut -d: -f1)"
+
+    readarray -d '' artifacts < <(find ${ARTIFACTS_DIR} -maxdepth 1 -type f -regextype posix-egrep -regex '.*\.zip$' -print0)
```
Comment (cpuguy83) [inline templates/sign.steps.windows.yml:77]:
```suggestion
    readarray -d '' artifacts < <(find "${ARTIFACTS_DIR}" -maxdepth 1 -type f -regextype posix-egrep -regex '.*\.zip$' -print0)
```

File: templates/sign.steps.windows.yml [hunk 4]
```diff
@@ -11,9 +10,13 @@ steps:
 
     tree ${ARTIFACTS_DIR}
 
-    readarray -d '' artifacts < <(find ${ARTIFACTS_DIR} -type f -regextype posix-egrep -regex '.*\.zip$' -print0)
-    for i in "${artifacts[@]}"; do
-      unzip -d $(dirname "$i") $i
+    readarray -d '' artifacts < <(find ${ARTIFACTS_DIR} -maxdepth 1 -type f -regextype posix-egrep -regex '.*\.zip$' -print0)
```
Comment (cpuguy83) [inline templates/sign.steps.windows.yml:13]:
```suggestion
    readarray -d '' artifacts < <(find "${ARTIFACTS_DIR}" -maxdepth 1 -type f -regextype posix-egrep -regex '.*\.zip$' -print0)
```

**Review by cpuguy83 — CHANGES_REQUESTED** [review]

File: queue_consumer.yaml [hunk 5]
```diff
@@ -121,14 +121,38 @@ stages:
...
               packageType: sdk
               version: 2.2.207
           - template: templates/sign.steps.yml
             parameters:
-              folderPath: "$(Pipeline.Workspace)/signed"
+              rootDir: "$(Pipeline.Workspace)/signed"
+          - bash: |
+              set -exu
+
+              win_dir="$SIGNED_DIR/windows"
```
Comment (cpuguy83) [inline queue_consumer.yaml:147]:
```suggestion
              win_dir="${SIGNED_DIR}/windows"
```

File: queue_consumer.yaml [hunk 6]
```diff
@@ -121,14 +121,38 @@ stages:
...
+              linux_dir="$SIGNED_DIR/linux"
```
Comment (cpuguy83) [inline queue_consumer.yaml:148]:
```suggestion
              linux_dir="${SIGNED_DIR}/linux"
```

File: queue_consumer.yaml [hunk 7]
```diff
@@ -121,14 +121,38 @@ stages:
...
+
+              rm -rf "$SIGNED_DIR/extracted"
```
Comment (cpuguy83) [inline queue_consumer.yaml:150]:
```suggestion
              rm -rf "${SIGNED_DIR}/extracted"
```

File: queue_consumer.yaml [hunk 8]
```diff
@@ -121,14 +121,38 @@ stages:
           - bash: |
               cp -r $(Pipeline.Workspace)/downloaded $(Pipeline.Workspace)/signed
             displayName: Copy downloaded files to sign
+          - bash: |
+              set -exu
+
+              win_dir="$SIGNED_DIR/windows"
+              linux_dir="$SIGNED_DIR/linux"
```
Comment (cpuguy83) [inline queue_consumer.yaml:128]:
```suggestion
              linux_dir="${SIGNED_DIR}/linux"
```

File: templates/sign.steps.windows.yml (hunk 1 above)
Comment (pmengelbert) [reply templates/sign.steps.windows.yml:17]: Did you mean this?
```suggestion
      dir="$ARTIFACTS_DIR/extracted/${base}"
```

File: templates/sign.steps.windows.yml (hunk 2 above)
Comment (pmengelbert) [reply templates/sign.steps.windows.yml:90]: Is this for aesthetics? A slash can't be part of a variable name, so there's no ambiguity to resolve with the braces.

File: templates/sign.steps.windows.yml (hunk 4 above)
Comment (pmengelbert) [reply templates/sign.steps.windows.yml]: done

File: queue_consumer.yaml (hunk 5 above)
Comment (pmengelbert) [reply queue_consumer.yaml]: done

File: queue_consumer.yaml (hunk 6 above)
Comment (pmengelbert) [reply queue_consumer.yaml]: done

File: queue_consumer.yaml (hunk 8 above)
Comment (pmengelbert) [reply queue_consumer.yaml]: done

**Review by cpuguy83 — APPROVED** [review]

**Pull request merged**